blocks/parser.js

```diff
@@ -37,16 +39,14 @@ export function getBlockAttributes( blockNode, blockSettings ) {
  * @return {Array}          Block list
  */
 export default function parse( content ) {
-	return grammarParse( content ).reduce( ( memo, blockNode ) => {
+	return grammarParse( content ).map( ( blockNode ) => {
 		const settings = getBlockSettings( blockNode.blockType );
```

Comment:
>The grammar already returns and `html` blockType if no comment is found

Yeah, I wasn't sure how I felt about that, vs. simply omitting the property if it isn't a block.

>For me it's ok to have a "special" block defined in all use cases (auto-registered) in the blocks module

This could work. It will likely add a dependency to `wp.element` from blocks and blocks would now include accompanying CSS.

A subtle side-effect to the logic in the visual editor render is that even if a block is registered, it's also gracefully handled if it defines neither `edit` nor `save`. I don't think we'd care to support this, so it probably shouldn't need to be graceful.